pkgs/development/libraries/libgweather/default.nix

```diff
@@ -69,20 +70,27 @@ stdenv.mkDerivation rec {
     "-Dc_args=-D_DARWIN_C_SOURCE"
   ];
 
+  # ensure build-time scripts like gen_locations_variant.py use the build gobject types.
+  env.GI_TYPELIB_PATH = "${lib.getLib buildPackages.glib}/lib/girepository-1.0";
```

Comment:
:/ I think this is a bug exposed by the splitting of glib from gobject-introspection

In libraries it's fine to do this since there's no `wrapGAppsHook`

I tried the following but it didn't work

```
substituteInPlace data/meson.build --replace-fail \
  "env: nomalloc," \
  "env: nomalloc + ['GI_TYPELIB_PATH=${lib.getLib buildPackages.glib}/lib/girepository-1.0'],"
```

Reply:
we could take inspiration from `patchShebangs --build`, and implement `wrapGApp --build`, then call that manually from here?

----
this `GI_TYPELIB_PATH` bit isn't so intrusive, but it will become a real problem once we tackle gtk apps using `blueprint-compiler`, which AFAICT need a build-time copy of _every_ runtime GIR file, unless we emulate the blueprint compiler itself.

Reply:
take a look at the new commit i just pushed though: it overrides GI_TYPELIB_PATH specifically for the binary which needs it instead of for the whole build process, as i think you were aiming to do above.